src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,61 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        List<String> lines = getStringFromFile(fromFileName);
+        Map<String, Integer> map = calculateAndFormReport(lines);
+        writeDataToFile(map, toFileName);
+    }
+
+    private List<String> getStringFromFile(String fromFileName) {
+        List<String> lines = new ArrayList<>();
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            while (bufferedReader.ready()) {
+                lines.add(bufferedReader.readLine());
+            }
+            return lines;
+        } catch (IOException e) {
+            throw new RuntimeException("Cannot read from .csv file " + fromFileName, e);
+        }
+    }
+
+    private Map<String, Integer> calculateAndFormReport(List<String> lines) {
+        int supply = 0;
+        int buy = 0;
+        int addingNumber;
+        Map<String, Integer> map = new LinkedHashMap<>();
+        for (String string : lines) {
+            addingNumber = Integer.parseInt(string.substring(string.indexOf(',') + 1));
+            if (string.contains("supply")) {
```

Comment:
create constants for supply, buy, result and comma symbol

Reply:
> create constants for supply, buy, result and comma symbol

Do you mean final class fields for supply, buy Strings and comma char symbol?